src/app/business/delfin/storages/storages.component.ts

```diff
@@ -0,0 +1,601 @@
+                    })
+                },
+                reject:()=>{}
+            })
+        }
+    }
+
+    deleteStorage(storage){
+
+        this.ds.deleteStorage(storage).subscribe(res=>{
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: 'Storage deleted successfully.'});
+            this.getAllStorages();
+        }, (error)=>{
+            this.msgs = [];
+            this.msgs.push({severity: 'error', summary: 'Error', detail: 'Error deleting Storage'});
+            console.log("Something went wrong. Could not delete storage", error);
+        });
+    }
+
+    showAlertSourceDialog(storage){
+        this.showRegisterAlertSourceForm = true;
+        this.v2cFields = false;
+        this.v3Fields = false;
+        this.selectedStorageId = storage['id']
+        this.registerAlertSourceForm.reset();
+    }
+
+    prepareFormDataArray(value){
+        let dataArr = {
+            version: value['version'],
+            username: value['username'],
+            engine_id: value['engine_id'],
+            security_level: value['security_level'],
+            auth_protocol: value['auth_protocol'],
+            auth_key: value['auth_key'],
+            privacy_protocol: value['privacy_protocol'],
+            privacy_key: value['privacy_key'],
+            host: value['host'],
+        };
+
+        if(value['community_string']){
+            dataArr['community_string'] = value['community_string'];
+        }
+
+        return dataArr;
+    }
+    setFormByVersion(version){
+        switch (version) {
+            case 'SNMPv2c':
+                    this.v2cFields = true;
+                    this.v3Fields = false;
+                break;
+            case 'SNMPv3':
+                    this.v2cFields = false;
+                    this.v3Fields = true;
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    registerAlertSource(value){
+        if(!this.registerAlertSourceForm.valid){
+            for(let i in this.registerAlertSourceForm.controls){
+                this.registerAlertSourceForm.controls[i].markAsTouched();
+            }
+            return;
+        }
+        let dataArr = this.prepareFormDataArray(value);
+
+        this.ds.registerAlertSource(this.selectedStorageId, dataArr).subscribe((res)=>{
+            this.msgs = [];
+            this.msgs.push({severity: 'success', summary: 'Success', detail: 'Storage device registered successfully.'});
+            let queryParams = {
+                "message": JSON.stringify({severity: 'success', summary: 'Success', detail: 'Storage device registered successfully.'})
+            };
+            this.router.navigate(['/delfin'], {queryParams: queryParams});
+        }, (error) =>{
+            this.msgs = [];
+            this.msgs.push({severity: 'error', summary: "Error", detail:"Something went wrong. Storage device could not be registered."});
```

Comment:
```suggestion
            this.msgs.push({severity: 'error', summary: "Error", detail:"Something went wrong. Storage device could not be registered for alert."});
```

Reply:
The message was a typo. I will change it. The correct message should be `Alert source could not be registered.`